File: webapp/src/stores/PauseAction.ts
export enum PauseType {
  TrainingPause = 0,
  OperationPause = 1
}

export class PauseAction {
  public ID: number;

  public UserID: number;

  public Type: PauseType;

  public StartTime: Date;

  public EndTime: Date;

  constructor(data: Record<string, any>) {
    this.setData(data);
  }

  public setData(data: Record<string, any>) {
    this.ID = data.ID;
    this.UserID = data.UserID;
    this.Type = data.Type;
    this.StartTime = data.StartTime
    this.EndTime = data.EndTime;
  }

  public getData(): Record<string,any> {
    return {
      ID: this.ID,
      UserID: this.UserID,
      Type: this.Type,
      StartTime: this.StartTime,
      EndTime: this.EndTime
    };
  }
}

File: models/pause_action.go
package models

import "time"

type PauseType uint8

const (
	TrainingPause  PauseType = 0
	OperationPause PauseType = 1
)

var PauseActionT = &PauseAction{}

type PauseAction struct {
	FameModel `gorm:"embedded_prefix:pa_"`
	UserID    uint64
	User      *User
	StartTime *time.Time
	EndTime   *time.Time
	Type      PauseType
}

// ColumnPrefix implements the gorm columnPrefixer interface
// and returns the column prefix
func (pa *PauseAction) ColumnPrefix() string {
	return "pa_"
}
